lms/djangoapps/grades/new/course_grade.py

```diff
@@ -24,8 +27,40 @@ class CourseGrade(object):
     def __init__(self, student, course, course_structure):
         self.student = student
         self.course = course
+        self.course_version = getattr(course, 'course_version', None)
+        self.course_edited_timestamp = getattr(course, 'subtree_edited_on', None)
         self.course_structure = course_structure
-        self.chapter_grades = []
+        self._percent = None
+        self._letter_grade = None
+        self.subsection_grade_factory = SubsectionGradeFactory(self.student, self.course, self.course_structure)
+
+    @classmethod
+    def load_persisted_grade(cls, user, course, course_structure):
+        """
+        Initializes a CourseGrade object, filling its members with persisted values from the database.
+
+        If the grading policy is out of date, recomputes the grade.
+
+        If no persisted values are found, returns None.
+        """
+        try:
+            persistent_grade = PersistentCourseGrade.read_course_grade(user.id, course.id)
+        except PersistentCourseGrade.DoesNotExist:
+            return None
+        course_grade = CourseGrade(user, course, course_structure)
+
+        current_grading_policy_hash = course_grade.get_grading_policy_hash(course.location, course_structure)
+        if current_grading_policy_hash != persistent_grade.grading_policy_hash:
+            course_grade.compute_and_update(read_only=False)
```

Comment:
Why not just return `None` in this case?  If the grading policy is out of date, then that means the persisted grade is out of date.  So returning `None` seems semantically correct.  Also, by calling `compute_and_update` here, the `read_only` value is not being honored by the higher callers.